agents/demo/demo.ipynb

```diff
@@ -0,0 +1,970 @@
+    "\n",
+    "- Create a service account with the appropriate roles and download the private key\n",
+    "- Use JupyterLab to upload the service account to your pod\n",
+    "- Set the path to your service account in the cell below and then execute it to activate the service account"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "KEY_FILE=\"/Users/cb/Downloads/kubeflow-rl-ec0f4f646339.json\"\n",
+    "!gcloud auth activate-service-account --key-file={KEY_FILE}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!gsutil mb -p {PROJECT} gs://{BUCKET}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!gcloud container clusters --project={PROJECT} --zone={ZONE} get-credentials {CLUSTER}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!kubectl create namespace {NAMESPACE}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Download and install ksonnet if needed"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!if ! [[ $(which ks) ]]; then mkdir -p ${HOME}/bin && curl -L -o ${HOME}/bin/ks \"https://github.com/ksonnet/ksonnet/releases/download/v0.8.0/ks-linux-amd64\" && chmod a+rx ${HOME}/bin/ks; fi"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If running on GCP (or possibly another Cloud) you probably need to create a key with credentials to use for your job"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "SECRET_FILE_NAME=\"secret.json\"\n",
+    "!kubectl create -n {NAMESPACE} secret generic {SECRET_NAME} --from-file={SECRET_FILE_NAME}={KEY_FILE}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Training"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The objective of the training phase is to learn the parameterization of our model that confers a high level of performance on the provided task. Here we'll launch and monitor a job."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Launching the TFJob"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We'll use [ksonnet](https://ksonnet.io/) to parameterize and apply a TFJob configuration (i.e. run a job). Here you can change the image to be a custom job image, such as one built and deployed with build.sh, or use the one provided here if you only want to change parameters. Below we'll display the templated job YAML for reference."
```

Comment:
I would like more explanation about what `kubeflow` is doing at each step, if the objective is to teach readers about more fundamental kubeflow-related concepts.

Would be good to link back to [kubeflow](https://github.com/kubeflow/kubeflow) docs, maybe the [user guide](https://github.com/kubeflow/kubeflow/blob/master/user_guide.md) as part of that.